ci/test/pipeline.template.yml

```diff
@@ -48,6 +48,22 @@ steps:
         agents:
           queue: builder-linux-aarch64
 
+      - id: build-x86_64-bazel
+        label: ":bazel: Build x86_64"
+        command: bin/ci-builder run stable bin/bazel-temp
+        depends_on: []
+        timeout_in_minutes: 60
+        agents:
+          queue: builder-linux-x86_64
+
+      - id: build-aarch64-bazel
+        label: ":bazel: Build aarch64"
+        command: bin/ci-builder run stable bin/bazel-temp
+        depends_on: []
+        timeout_in_minutes: 60
+        agents:
+          queue: builder-linux-aarch64
```

Comment:
Does this mean we'll be paying double for builds in CI? Or do we expect caching to make a meaningful dent here?

Reply:
![Screenshot 2024-07-23 at 23 38 08](https://github.com/user-attachments/assets/2c0942bc-be16-4caf-800c-71698aab0f1b)
The Bazel build seems much cheaper, so this is not as bad as it seems.

Reply:
Caching should make a pretty meaningful dent, and once we get the `toolchains` repo setup and https://github.com/MaterializeInc/materialize/pull/28370 merged they should be even faster.

More than happy to keep this in nightly for now if cost is a concern!